src/blockencodings.h

```diff
@@ -105,16 +103,22 @@ struct PrefilledTransaction {
     uint16_t index;
     CTransactionRef tx;
 
-    ADD_SERIALIZE_METHODS;
+    template<typename Stream>
+    void Serialize(Stream& s) const
+    {
+        s << COMPACTSIZE(index) << Using<TransactionCompression>(tx);
+    }
 
-    template <typename Stream, typename Operation>
-    inline void SerializationOp(Stream& s, Operation ser_action) {
-        uint64_t idx = index;
-        READWRITE(COMPACTSIZE(idx));
-        if (idx > std::numeric_limits<uint16_t>::max())
+    template<typename Stream>
+    void Unserialize(Stream& s)
```

Comment:
In commit "Convert blockencodings.h to new serialization framework" (d46dfeccaf1992ba92bff7e0d4988ae3f66b327a)

There should be no need to have custom serialize and deserialize methods, and do overflow checking with a temporary variable now that `CompactSizeFormatter` has the same checking. The following seems to be sufficient:

```c++
SERIALIZE_METHODS(PrefilledTransaction, obj) { READWRITE(COMPACTSIZE(obj.index), obj.tx); }
```

Reply:
Nice, done!